vunerability upgrade PR

junaidwarsivd opened the pull request.

junaidwarsivd committed:
did vunerability upgrade

dubdabasoduba committed:
Merge branch 'master' into vunerability-upgrades

dubdabasoduba committed:
Merge branch 'master' into vunerability-upgrades

dubdabasoduba commented on opensrp-anc/build.gradle:
```diff
@@ -251,11 +251,11 @@ dependencies {
-    implementation 'org.jeasy:easy-rules-core:3.3.0'
-    implementation 'org.jeasy:easy-rules-mvel:3.3.0'
-    implementation("com.itextpdf:itext7-core:7.1.12")
+    implementation 'org.jeasy:easy-rules-core:3.4.0'
+    implementation 'org.jeasy:easy-rules-mvel:3.4.0'
```
@bonfaceshisakha pointed out that this needs to be version 4.0.0

junaidwarsivd committed:
jeasy rules updated

junaidwarsivd committed:
Merge branch 'vunerability-upgrades' of https://github.com/opensrp/opensrp-client-anc into vunerability-upgrades

ekigamba committed:
Fix MVEL rules not working with jeasy-mvel 4.1.0

- Update jackson-dataformat-yaml to 2.14.0 to fix security vulnerabilities
- Update native-form to fix MVEL rules not working for some forms such as registration forms

ekigamba commented on opensrp-anc/src/main/java/org/smartregister/anc/library/YamlRuleDefinitionReaderExt.java:
```diff
@@ -0,0 +1,32 @@
+/**
+ * Created by Ephraim Kigamba - nek.eam@gmail.com on 22-11-2022.
+ */
+public class YamlRuleDefinitionReaderExt extends YamlRuleDefinitionReader {
```
This can be removed and we can use the one on native form

ekigamba commented on opensrp-anc/src/main/java/org/smartregister/anc/library/helper/AncRulesEngineFactory.java:
```diff
@@ -45,4 +49,36 @@ protected Facts initializeFacts(Facts facts) {
     }
+
+    @Override
+    public void beforeExecute(Rule rule, Facts facts) {
```
We can possibly remove this and use what is on native form

ekigamba commented on opensrp-anc/src/main/java/org/smartregister/anc/library/helper/AncRulesEngineHelper.java:
```diff
@@ -53,14 +55,33 @@ public class AncRulesEngineHelper extends RulesEngineHelper {
     private final Map<String, Rules> ruleMap;
     private JSONObject mJsonObject = new JSONObject();
-    private final YamlRuleDefinitionReader yamlRuleDefinitionReader = new YamlRuleDefinitionReader();
+    private final YamlRuleDefinitionReaderExt yamlRuleDefinitionReader = new YamlRuleDefinitionReaderExt();
     private final MVELRuleFactory mvelRuleFactory = new MVELRuleFactory(yamlRuleDefinitionReader);
 
     public AncRulesEngineHelper(Context context) {
         this.context = context;
         this.inferentialRulesEngine = new InferenceRulesEngine();
         RulesEngineParameters parameters = new RulesEngineParameters().skipOnFirstAppliedRule(true);
         this.defaultRulesEngine = new DefaultRulesEngine(parameters);
+       /* ((DefaultRulesEngine) this.defaultRulesEngine).registerRuleListener(new RuleListener() {
+            @Override
+            public void beforeExecute(Rule rule, Facts facts) {
+                Timber.e("Putting facts in beforeExecute");
+                facts.put("facts", facts);
+            }
+
+            @Override
+            public void onSuccess(Rule rule, Facts facts) {
+                Timber.e("Putting facts in onSuccess");
+                facts.remove("facts");
+            }
+
+            @Override
+            public void onFailure(Rule rule, Facts facts, Exception exception) {
+                Timber.e("Putting facts in onFailure");
+                facts.remove("facts");
+            }
+        });*/
```
We can discard these changes

junaidwarsivd committed:
uptadated the facts population in rulesEnginehelper

junaidwarsivd committed:
fixed the AncRulesEngineFactory and AncRulesEngineHelper for the not showing the overview and previous contacts in the contacts sections

junaidwarsivd committed:
fixed the AncRulesEngineFactory and AncRulesEngineHelper for the not showing the overview and previous contacts in the contacts sections

junaidwarsivd committed:
updated the vunerability changes

junaidwarsivd committed:
removed dependencies that were already included in native forms
